psutil/_psaix.py

```diff
@@ -386,8 +379,10 @@ def _proc_cred(self):
     def name(self):
         if self.pid == 0:
             return "swapper"
-        # note: this is limited to 15 characters
-        return self._proc_name_and_args()[0].rstrip("\x00")
+        # pr_fname is PRFNSZ (16) characters long
+        # if the name is longer than 16 characters, pr_fname will not
+        # be NULL-terminated, so we truncate it here
+        return cext.proc_name(self.pid, self._procfs_path)[:16]
```

Comment:
This is what we had, but it didn't work (see the comment - there wouldn't be a \x00).

Reply:
what would there be instead? spaces?

Reply:
Since we're dealing with a non-NULL terminated string, we will just continue to read the next field after the name, which happens to be `pr_psargs`.
For a process named `abcdefghijklmnopq` (17 characters), we would read `abcdefghijklmnop./abcdefghijklmnopq`.
`psutil.__init__` has logic that reads the name from `cmdline` if the name is longer or equal to 15 characters, but it does that with `startswith`, so we have to do the truncation to make it work.

Reply:
I understand now